webpack/JobWizard/steps/CategoryAndTemplate/CategoryAndTemplate.test.js

```diff
@@ -1,52 +1,149 @@
-import { testComponentSnapshotsWithFixtures } from '@theforeman/test';
-import { CategoryAndTemplate } from './CategoryAndTemplate';
-
-const baseProps = {
-  setJobTemplate: jest.fn(),
-  selectedTemplateID: 190,
-  setCategory: jest.fn(),
-};
-const fixtures = {
-  'renders with props': {
-    ...baseProps,
-    jobCategories: [
-      'Commands',
-      'Ansible Playbook',
-      'Ansible Galaxy',
-      'Ansible Roles Installation',
-    ],
-    jobTemplates: [
-      {
-        id: 190,
-        name: 'ab Run Command - SSH Default clone',
-        job_category: 'Commands',
-        provider_type: 'SSH',
-        snippet: false,
-      },
-      {
-        id: 168,
-        name: 'Ansible Roles - Ansible Default',
-        job_category: 'Ansible Playbook',
-        provider_type: 'Ansible',
-        snippet: false,
-      },
-      {
-        id: 170,
-        name: 'Ansible Roles - Install from git',
-        job_category: 'Ansible Roles Installation',
-        provider_type: 'Ansible',
-        snippet: false,
-      },
-    ],
-    selectedCategory: 'I am a category',
-  },
-  'render with error': {
-    ...baseProps,
-    errors: { allTemplatesError: 'I have an error' },
-  },
-};
-
-describe('CategoryAndTemplate', () => {
-  describe('rendering', () =>
-    testComponentSnapshotsWithFixtures(CategoryAndTemplate, fixtures));
+import React from 'react';
+import { Provider } from 'react-redux';
+import configureMockStore from 'redux-mock-store';
+import { fireEvent, screen, render, act } from '@testing-library/react';
+import * as api from 'foremanReact/redux/API';
+import { JobWizard } from '../../JobWizard';
+import * as selectors from '../../JobWizardSelectors';
+import { jobTemplate, jobTemplateResponse } from '../../__tests__/fixtures';
+
+jest.spyOn(api, 'get');
+jest.spyOn(selectors, 'selectJobTemplate');
+jest.spyOn(selectors, 'selectJobTemplates');
+jest.spyOn(selectors, 'selectJobCategories');
+jest.spyOn(selectors, 'selectCategoryError');
+jest.spyOn(selectors, 'selectAllTemplatesError');
+jest.spyOn(selectors, 'selectTemplateError');
```

Comment:
how about those?

Reply:
left only `select___Error` as they are the only relevant ones for the tst